git_connector.py

```diff
@@ -61,10 +61,13 @@ def initialize(self):
         called.
         """
 
+        config = self.get_config()
         self.app_state_dir = Path(self.get_state_dir())
-
-        if self.get_action_identifier() == "configure_ssh":
-            return phantom.APP_SUCCESS
+        self.username = config.get(consts.GIT_CONFIG_USERNAME)
+        self.password = config.get(consts.GIT_CONFIG_PASSWORD)
+        self.branch_name = config.get(consts.GIT_CONFIG_BRANCH_NAME, 'master')
+        self.repo_name = config.get(consts.GIT_CONFIG_REPO_NAME)
```

Comment:
Based on the latest code changes, we are getting repo name directly from the config parameter which causes error in test connectivity and all the other actions which uses the `self.repo_name` variable. 
As `self.repo_name` variable is an optional variable and if user does not pass any value in it will take `None` value and the actions get failed. Instead of which we want it to take the repo name from repo uri passed in config parameter.

Expected Behaviour:
1) If no repo name is passed in the config parameter it should take the repo name from repo uri.
2) If repo name passed in config parameter then it should use the repo name provided in config parameter.

Reply:
Currently `set_repo_attributes` method is getting called in clone repo and delete repo actions.
Seems like following actions `_delete_file, _git_pull, _add_file, _git_push, _git_commit, _update_file, _git_status` require `repo name` which is defined in `set_repo_attributes` method. Also, test asset connection require modified repo uri which is also defined in `set_repo_attributes` method. This is why it would be better to initialize all the variables of set_repo_attributes in initialize() itself. Could you please check this?

Reply:
Hello @jrm16020,
What we want to highlight here is,

initialize() has self.repo_name = config.get(consts.GIT_CONFIG_REPO_NAME) which would be used in the following actions: git pull, git commit, git push, git status, add file, delete file and update file. Due to this, here the repo_name would just be the name we configured in the asset.

But the other actions: clone repo, delete clone use self.repo_name = config.get(consts.GIT_CONFIG_REPO_NAME, f"{temp_repo_name}_{self.branch_name}") as you have called _set_repo_attributes() from them due to which they will use repo_name as <name_of_repo>_<name_of_branch>.

This will lead to following issues: Cloning a repo would add repo with name <repo_name>_<branch_name>. When you try deleting a clone, it would search for the correct repo name in the path and delete successfully as the branch name is appended after repo name while cloning. Thus, clone repo and delete clone will work fine here. But, all the other actions that help us making changes in the cloned repo will fail as they will use just the repo_name(without the branch name appended) in their implementation. Due to this they will not find the repo in the path and would fail. This is just because they use the self.repo_name initialized in the initialize() function. If they might have used the self.repo_name from the _set_repo_attributes() function, they would have got the correct repo path and would have passed the actions.

Thus the initialization of self.repo_name variable at two different places in two different ways is causing an issue even when this variable is used in the same way in all the actions.

Reply:
@jrm16020 Thanks for the changes.
As we are using the `_set_repo_attributes` method in most of the actions, it seems like we can call the specific method in initialize block instead of calling it in all the action blocks.
In case you don't want to add the call of `_set_repo_attributes` method in initialize block, we need to add the `_set_repo_attributes` method call in test asset connectivity actions as the action requires modified_repo_uri which is in the `_set_repo_attributes` method.